### DISCUSSION TITLE
use network drive with X_VCPKG_ASSET_SOURCES without mapping it as a drive

### DISCUSSION THREAD

**Discussion by AbdulAbCellera** [microsoft/vcpkg#29361] [Q&A]
I have a network-shared folder with FQDN myfs.hi.com/projs. I dont want to mount it, and I want to use it as the asset source. My issue is that I could not figure out the syntax for that.

### mounted worked
`set(ENV{X_VCPKG_ASSET_SOURCES} "x-azurl,file:///Z:/asset/,,readwrite")`

### regular dir worked
`set(ENV{X_VCPKG_ASSET_SOURCES} "x-azurl,file:///$ENV{USERPROFILE}/asset/,,readwrite")`

### fails
`set(ENV{X_VCPKG_ASSET_SOURCES} "x-azurl,file:///myfs.hi.com/projs/asset/,,readwrite")`
`set(ENV{X_VCPKG_ASSET_SOURCES} "x-azurl,file:///\\\\myfs.hi.com/projs/asset/,,readwrite")`
`set(ENV{X_VCPKG_ASSET_SOURCES} "x-azurl,file://///myfs.hi.com/projs/asset/,,readwrite")`

I use the drive for binary caching and it works (notice the slashes)
`set(VCPKG_DEFAULT_BINARY_CACHE "files,\\\\myfs.hi.com\projs\asset,readwrite")`

I see this error

```
Downloading https://github.com/Kitware/CMake/releases/download/v3.25.1/cmake-3.25.1-windows-i386.zip
  warning: failed to store back to mirror:
  error: curl failed to put file to file://myfs.hi.com/projs/asset/4a84abdb53781178a99046aa09cf6a6b2bfe159d6b
6bbdd2449839d2f50b872b4cc783c44774343b6122ec9400bcfdec65fffd4a246575ba55d
178bc180f6adc with exit code 0 and http code 302.
```

**Comment by JackBoosY** [comment]
Please fill an issue instead of this.

**Reply by AbdulAbCellera** [reply]
I have a doubt that it is an issue. It is more of am I using it correctly? I know that it is only a shared network drive the does not support https and anything fancy. 

If you still think I should file it as issue let me know.

**Reply by JackBoosY** [reply]
Hmm, maybe we should improve the expression parser.

**Comment by Gabboxl** [comment]
Hi, just had this problem simply because I didn't know how the file:// protocol works. To access network drives from root, the correct syntax is `file://networkdrive/folder`
